### PR TITLE
Climber

### PR DESCRIPTION
**Pull request by Knight4**

### CODE REVIEW

File: src/main/java/org/team199/robot2020/subsystems/Climber.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        winchMotor.getEncoder().setPositionConversionFactor(Constants.Climber.WINCH_CONVERSION_FACTOR);
+        winchMotor.getEncoder().setPosition(Constants.Climber.WINCH_START_HEIGHT);
+    }
+
+    public void runLift(double speed) {
+        liftMotor.set(speed);
+    }
+
+    public void runWinch(double speed) {
+        winchMotor.set(Math.abs(speed));
```
Comment (SaltyJoke) [inline src/main/java/org/team199/robot2020/subsystems/Climber.java:23]: What about when we reset the winch between matches?

File: src/main/java/org/team199/robot2020/commands/AdjustClimber.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  }
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    if (controller.getRawButton(Constants.OI.Controller.ADJUST_CLIMBER_UP_BUTTON)) { // up
+      climber.runLift(Constants.Climber.LIFT_KEEP_SPEED + Constants.Climber.LIFT_ADJUST_SPEED);
+    } else if (controller.getRawButton(Constants.OI.Controller.ADJUST_CLIMBER_DOWN_BUTTON)){ // down
+      climber.runLift(Constants.Climber.LIFT_KEEP_SPEED - Constants.Climber.LIFT_ADJUST_SPEED);
+    }
```
Comment (SaltyJoke) [inline src/main/java/org/team199/robot2020/commands/AdjustClimber.java:42]: Make sure it doesn't move when no buttons are pressed.

File: src/main/java/org/team199/robot2020/commands/DeployClimber.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  @Override
+  public void initialize() {
+    climber.runLift(Constants.Climber.LIFT_DEPLOY_SPEED);
+    climber.runWinch(Constants.Climber.WINCH_DEPLOY_SPEED);
+  }
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+      if (climber.getWinchHeight() >= 0) {
```
Comment (SaltyJoke) [inline src/main/java/org/team199/robot2020/commands/DeployClimber.java:35]: When will winch height be 0?

File: src/main/java/org/team199/robot2020/subsystems/Climber.java (hunk 1 above)
Comment (kevinzwang) [reply src/main/java/org/team199/robot2020/subsystems/Climber.java:23]: drill with hex adapter

File: src/main/java/org/team199/robot2020/commands/AdjustClimber.java (hunk 2 above)
Comment (kevinzwang) [reply src/main/java/org/team199/robot2020/commands/AdjustClimber.java:42]: you're right

File: src/main/java/org/team199/robot2020/commands/DeployClimber.java (hunk 3 above)
Comment (kevinzwang) [reply src/main/java/org/team199/robot2020/commands/DeployClimber.java:35]: negative: winding out, positive: winding in. in Climber.java, I set the climber encoder's position to -90 when it's initialized

File: src/main/java/org/team199/robot2020/commands/AdjustClimber.java (hunk 2 above)
Comment (kevinzwang) [reply src/main/java/org/team199/robot2020/commands/AdjustClimber.java:42]: fixed

**Pull request merged**